app/controllers/auth/sso_controller.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Auth
+  class SsoController < Api::V1::BaseController
+    DEFAULT_SESSION_DURATION = 8.hours
+
+    before_action :require_ee!
+    before_action :handle_callback_error,
+      only: %i[callback]
+
+    skip_verify_authorized
+
+    def callback
+      code = request.query_parameters[:code]
+
+      # redeem the callback authentication code for a user profile
+      profile = Keygen::EE::SSO.redeem_code(code:)
+
+      # lookup the account for the user's org
+      account = Account.where.not(sso_organization_id: nil) # sanity-check
+                             .find_by(
+                               sso_organization_id: profile.organization_id,
+                             )
+
+      unless account.present?
+        Keygen.logger.warn { "[sso] failed to find account: profile_id=#{profile.id.inspect} organization_id=#{profile.organization_id.inspect}" }
+
+        raise Keygen::Error::InvalidSingleSignOnError.new('failed to find account', code: 'SSO_INVALID_ACCOUNT')
+      end
+
+      # verify the user's email domain matches one of the account's domains
+      unless account.sso_for?(profile.email)
```

Comment:
Add config for allowing external users i.e. those that don't match a configured domain but came from WorkOS?